os/mm/mm_heap/mm_heapinfo.c

```diff
@@ -282,5 +321,113 @@ void heapinfo_exclude_stacksize(void *stack_ptr)
 
 	ASSERT(rtcb);
 	rtcb->curr_alloc_size -= node->size;
+
+#ifdef CONFIG_HEAPINFO_GROUP
+	int check_idx;
+	int group_num;
+	struct mm_heap_s *heap = mm_get_heap_info();
+
+	for (check_idx = 0; check_idx <= heap->max_group; check_idx++) {
+		if (node->pid == group_info[check_idx].pid) {
+			group_num = group_info[check_idx].group;
+			heap->group[group_num].curr_size -= node->size;
+			break;
+		}
+	}
+#endif
+}
+
+#ifdef CONFIG_HEAPINFO_GROUP
+/****************************************************************************
+ * Name: heapinfo_update_group_info
+ *
+ * Description:
+ * when create or release task/thread, check that the task/thread is 
+ * in group list
+ ****************************************************************************/
+void heapinfo_update_group_info(pid_t pid, int group, int type)
+{
+	int info_idx;
+	struct tcb_s *tcb;
+	struct mm_heap_s *heap = mm_get_heap_info();
+	switch (type) {
+	case HEAPINFO_INIT_INFO:
+		for (info_idx = 0; info_idx < CONFIG_MAX_TASKS; info_idx++) {
+			group_info[info_idx].pid = pid;
+			group_info[info_idx].group = group;
+			group_info[info_idx].stack_size = 0;
+		}
+		break;
+	case HEAPINFO_ADD_INFO:
+		if (group > heap->max_group) {
+			heap->max_group = group;
+		}
+		for (info_idx = 0; info_idx < CONFIG_MAX_TASKS; info_idx++) {
+			if (group_info[info_idx].pid <= 0) {
+				group_info[info_idx].pid = pid;
+				group_info[info_idx].group = group;
+				tcb = sched_gettcb(pid);
+				group_info[info_idx].stack_size = tcb->adj_stack_size;
+				heapinfo_update_group(tcb->adj_stack_size, pid);
+				break;
+			}
+		}
+		break;
+	case HEAPINFO_DEL_INFO:
+		for (info_idx = 0; info_idx < CONFIG_MAX_TASKS; info_idx++) {
+			if (pid == group_info[info_idx].pid) {
+				heapinfo_update_group((-1) * group_info[info_idx].stack_size, pid);
+
+				group_info[info_idx].pid = -1;
+				group_info[info_idx].group = -1;
+				group_info[info_idx].stack_size = 0;
+				break;
+			}
+		}
+		break;
+	default:
+		break;
+	}
+}
+
+/****************************************************************************
+ * Name: heapinfo_check_group_list
+ *
+ * Description:
+ * check that task/thread is in group list
+ ****************************************************************************/
+void heapinfo_check_group_list(pid_t pid, char *name)
+{
+	char *thread_list = CONFIG_HEAPINFO_PER_GROUP;
+
+	int group_num = 0;
+
+	char *name_start;
+	char *name_end;
+	int name_length;
+	name_start = name_end = thread_list;
+
+	while (*name_start != '\0') {
+		if (*name_end == '/' || *name_end == ',' || *name_end == '\0') {
+			name_length = name_end - name_start;
+			if (name_length == strlen(name)) {
+				if (strncmp(name_start, name, name_end - name_start) == 0) {
```

Comment:
strncmp basically consumes too much burden.
How about convert task/thread name into hash value, and just compare the value on here.
Hash value can be calculated on runtime, and also on build time